CODIGO/BabelUI.bas

```diff
@@ -0,0 +1,391 @@
+            ConvertMouseButton = kButton_None
+    End Select
+End Function
+
+Public Function CheckAndSetBabelUIUsage() As Boolean
+    UseBabelUI = Val(GetSetting("OPCIONES", "UseExperimentalUI"))
+    CheckAndSetBabelUIUsage = UseBabelUI
+End Function
+
+Public Function GetMainHwdn() As String
+    If UseBabelUI Then
+        GetMainHwdn = frmBabelLogin.hwnd
+    Else
+        GetMainHwdn = frmConnect.hwnd
+    End If
+    
+End Function
+
+Public Sub InitializeUI(ByVal Width As Long, ByVal Height As Long, ByVal pixelSize As Long)
+On Error GoTo InitializeUI_Err
+    If BabelInitialized Then Exit Sub
+100 Dim initSuccess As Boolean
+102 UITexture.Height = Height
+104 UITexture.Width = Width
+    Dim Settings As BABELSETTINGS
+    Settings.Height = Height
+    Settings.Width = Width
+#If Compresion = 1 Then
+    Settings.Compresed = 1
+#End If
+#If DEBUGGING = 1 Or Developer = 1 Then
+    Settings.EnableDebug = 1
+#End If
+106 initSuccess = InitializeBabel(Settings)
+108 UITexture.TextureHeight = NextPowerOf2(Height)
+110 UITexture.TextureWidth = NextPowerOf2(Width)
+112 ReDim UITexture.ImageBuffer(UITexture.Height * UITexture.Width * pixelSize)
+114 UITexture.pixelSize = pixelSize
+116 Set UITexture.Texture = SurfaceDB.CreateTexture(UITexture.TextureWidth, UITexture.TextureHeight)
+118 BabelInitialized = True
+    Call RegisterCallbacks(AddressOf LoginCB, AddressOf CloseClientCB, AddressOf BabelUI.CreateAccount, AddressOf SetHostCB, AddressOf ValidateCodeCB, AddressOf ResendValidationCodeCB, AddressOf RequestPasswordResetCB, AddressOf RequestNewPasswordCB, AddressOf SelectCharacterPreviewCB, AddressOf LoginCharacterCB, AddressOf ReturnToLoginCB, AddressOf CreateCharacterCB)
+    Exit Sub
+InitializeUI_Err:
+    Call RegistrarError(Err.Number, Err.Description, "BabelUI.InitializeUI", Erl)
+End Sub
+
+Public Sub InitializeInspectorUI(ByVal Width As Long, ByVal Height As Long)
+On Error GoTo InitializeInspectorUI_Err
+100 Dim initSuccess As Boolean
+102 Call CreateDebugWindow(Width, Height)
+104 DebugUITexture.Height = Height
+106 DebugUITexture.Width = Width
+108 DebugUITexture.TextureHeight = NextPowerOf2(Height)
+110 DebugUITexture.TextureWidth = NextPowerOf2(Width)
+112 ReDim DebugUITexture.ImageBuffer(DebugUITexture.Height * DebugUITexture.Width * 4)
+114 DebugUITexture.pixelSize = 4
+116 Set DebugUITexture.Texture = SurfaceDB.CreateTexture(DebugUITexture.TextureWidth, DebugUITexture.TextureHeight)
+    If DebugUITexture.Texture Is Nothing Then
+118    Call RegistrarError(102, "texture undefined ", "BabelUI.InitializeInspectorUI", 102)
+       Exit Sub
+    End If
+    DebugInitialized = True
+    Exit Sub
+InitializeInspectorUI_Err:
+    Call RegistrarError(Err.Number, Err.Description, "BabelUI.InitializeInspectorUI", Erl)
+End Sub
+
+Public Sub DrawUITexture(ByRef TextureInfo As t_UITexture)
+On Error GoTo DrawTexture_Err
+    With TextureInfo
+100     If Not .Texture Is Nothing Then
+110         If .Texture Is Nothing Then
```

Comment:
```
 If Not .Texture Is Nothing Then
   If .Texture Is Nothing Then
```
The second if can be removed, surely if the first is true the second one cannot be true?